flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateCheckpointWriter.java

```diff
@@ -296,6 +298,9 @@ private void runWithChecks(RunnableWithException r) throws Exception {
             r.run();
         } catch (Exception e) {
             fail(e);
+            if (!findThrowable(e, IOException.class).isPresent()) {
```

Comment:
Was it your intention to ignore exceptions with `IOException` at any position in the stack? Shouldn't we ignore only top level `IOException`. 

BTW, I find the logic a bit confusing. What is so different between `IOException`s and all the other? Shouldn't we treat them the same? Could you elaborate on cases when we don't want to ignore the exception?

Reply:
Yes, my intention was to not fail the job (but only checkpoint) if the underlying FS abstraction wraps this into RuntimeException for example (it shouldn't).
The reasoning behind IOExceptions failing only the current checkpoint is that they can be transient (e.g. network failure); other exceptions usually aren't transient (e.g. programmatic errors) and therefore should fail the job.